geoh5py/objects/grid_object.py

```diff
@@ -142,10 +85,79 @@ def mask_by_extent(
 
         Applied to object's centroids.
         """
-        if self.extent is None or not box_intersect(self.extent, extent):
+        if not box_intersect(self.extent, extent):
             return None
 
-        if self.centroids is not None:
-            return mask_by_extent(self.centroids, extent, inverse=inverse)
+        return mask_by_extent(self.centroids, extent, inverse=inverse)
+
+    @property
+    def n_cells(self) -> int:
+        """
+        Total number of cells
+        """
+        return int(np.prod(self.shape))
+
+    @property
+    def rotation(self) -> float:
+        """
+        Clockwise rotation angle (degree) about the vertical axis.
+        """
+        return self._rotation
+
+    @rotation.setter
+    def rotation(self, value: np.ndarray | Number):
+        if isinstance(value, Number):
```

Comment:
Because it stored on geoh5 as an array

Reply:
type(2.0) == type(np.r_[2].astype(float).item())

You are converting the array back to float when assigning to self._rotation.
I'm ok if it can take a numpy as input, it just seems overly complex for a float setter, even if the setter can accept an array of shape 1.

You can convert directly a 1d array with float(np.array([2]))